src/main/java/Parser.java

```diff
@@ -0,0 +1,56 @@
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+public class Parser {
+    public static LocalDateTime parseDateTime(String dateTime) throws DukeException {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd-MM-yyyy HH:mm");
+
+        try {
+            return LocalDateTime.parse(dateTime, formatter);
+        } catch (DateTimeParseException e) {
+            throw new DukeInvalidArgumentException("Please enter a valid datetime: dd-MM-yyyy HH:mm");
+        }
+    }
+
+    public static void processCommand(String userCommand, Storage storage, TaskList tasks) throws DukeException {
+
+        if (userCommand.isEmpty()) {
+            throw new DukeInvalidCommandException("Please enter a command");
+        }
+
+        String command = userCommand.split(" ", 2)[0];
+
+        // Single word commands
+        if (userCommand.equals("list")) {
+            tasks.printTaskList();
+            return;
+        }
+
+        // Multi-word commands
+        switch (command) {
```

Comment:
i like that you implement switch case instead of multiple if-else statements which i see more often. i like how concise it makes your code look